src/boringssl/gen_build_yaml.py

```diff
@@ -14,20 +14,20 @@
 # limitations under the License.
 
 from __future__ import print_function
+from collections import OrderedDict
+
+import json
 import shutil
 import sys
 import os
 import yaml
 
-sys.dont_write_bytecode = True
-
-boring_ssl_root = os.path.abspath(
-    os.path.join(os.path.dirname(sys.argv[0]),
-                 '../../third_party/boringssl-with-bazel/src'))
-sys.path.append(os.path.join(boring_ssl_root, 'util'))
-
 try:
-    import generate_build_files
+    sources_path = os.path.abspath(
```

Comment:
Why is this being changed?

I don't like that these changes are hiding under the disguise of just upgrading boringssl version.
Please explain why these changes were necessary and what's the design behind them.
This really seems the be the most significant change of this PR so I'd actually prefer this:
1. perform a routing boringssl update according to our instructions (https://github.com/grpc/grpc/tree/master/third_party#updating-third_partyboringssl-with-bazel)

2. change the way the boringssl sources are extracted, but do that in a small separate PR so that it's clear what's being done and why.

Reply:
To give you context, the path that led me here is:
1.  python gRPC is being compiled without ASM optimizations (https://github.com/grpc/grpc/pull/23483)
2.  the way this code currently hacks around the boringssl build system doesn't work for capturing ASM files.  When I attempted a small change to BoringSSL to capture the ASM files, the BoringSSL team [pushed](https://boringssl-review.googlesource.com/c/boringssl/+/42164) for cleaning up technical debt.

I didn't intend to hide this change, I noted why I thought it might be worthwhile to include this in [a comment on this PR](https://github.com/grpc/grpc/pull/23528#issuecomment-660335200).  Is it OK if Include this change in the [PR to turn on ASM for python gRPC](https://github.com/grpc/grpc/pull/23483)?  Or would you prefer two distinct PRs (1  with this change and 1 for the python changes)?

Reply:
I think the best way is:
1. PR1: upgrade boringssl according to our docs as usual (a routine PR), don't change the way boringssl sources are processed.
2. PR2: once one the right boringssl version, suggest how processing of the source file should be changed. This change needs to be reviewed very carefully to make sure nothing breaks, so there should not be any boilerplate in this change.
3. PR3: enable asm in python (initially based off of PR2 to show that it works, then rebased once PR2 is in).

Reply:
Thanks. File reverted.  I'll open up PR2 once this one gets merged.